pkgs/by-name/bu/bugwarrior/package.nix

```
@@ -0,0 +1,93 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  taskwarrior3,
+  writableTmpDirAsHomeHook,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "bugwarrior";
+  version = "2.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "GothenburgBitFactory";
+    repo = "bugwarrior";
+    tag = version;
+    hash = "sha256-VuHTrkxLZmQOxyig2krVU9UZDDbLY08MfB9si08lh3E=";
+  };
+
+  build-system = with python3Packages; [ setuptools ];
+
+  dependencies =
+    with python3Packages;
+    [
+      click
+      dogpile-cache
+      jinja2
+      lockfile
+      pydantic
+      python-dateutil
+      pytz
+      requests
+      taskw
+      tomli
+    ]
+    ++ pydantic.optional-dependencies.email;
+
+  optional-dependencies = {
+    bts = with python3Packages; [ debianbts ];
+    bugzilla = with python3Packages; [ python-bugzilla ];
+    gmail = with python3Packages; [
+      google-api-python-client
+      google-auth-oauthlib
+    ];
+    jira = with python3Packages; [ jira ];
+    keyring = with python3Packages; [ keyring ];
+    trac = with python3Packages; [ offtrac ];
+  };
+
+  nativeCheckInputs = [
+    writableTmpDirAsHomeHook
+  ]
+  ++ (with python3Packages; [
+    pytestCheckHook
+    docutils
+    pytest-subtests
+    responses
+    ruff
+    sphinx
+    sphinx-click
+    sphinx-inline-tabs
+    importlib-metadata
+    taskwarrior3
+  ])
+  ++ lib.concatAttrValues optional-dependencies;
+  disabledTestPaths = [
+    # Optional dependencies for these services aren't packaged.
+    "tests/test_kanboard.py"
+    "tests/test_phab.py"
+    "tests/test_todoist.py"
+  ];
+  disabledTests = [
+    # Requires ini2toml dependency, which isn't packaged.
+    "TestIni2Toml"
+    # Import services for which the optional dependencies aren't packaged.
+    "TestValidation"
+    "ExampleTest"
+    "TestServices"
+  ];
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
https://github.com/NixOS/nixpkgs/issues/208242